assets/src/scripts/components/embed.spec.js

```diff
@@ -0,0 +1,23 @@
+const { select } = require('d3-selection');
+
+const { attachToNode } = require('./embed');
+
+
+describe('embed component', () => {
+    beforeEach(() => {
+        const elem = select('body')
+            .append('div')
+                .attr('id', 'component')
+                .classed('wdfn-component', true)
+                .attr('data-component', 'embed');
+        attachToNode(null, elem.node());
+    });
+
+    afterEach(() => {
+        select('#component').remove();
+    });
+
+    it('should prepend sharing icon', () => {
+        expect(select('.fa-share').size()).toBe(1);
+    });
+});
```

Comment:
We might want to consider testing of event handlers. I have artificially generated events in Jasmine tests: https://github.com/NWQMC/WQP_UI/blob/01863fd28fadbadf8de5fb094cc51754ea09429b/test/js/portal_ui/leafletControls/FeatureSourceSelectControlSpec.js#L73